pkg/replicate/scheme.go

```diff
@@ -192,7 +198,11 @@ func (rs *replicationScheme) execute(ctx context.Context) error {
 	}
 
 	for id, meta := range metas {
-		if rs.blockFilter(meta) {
+		_, err := rs.fromBkt.ReaderWithExpectedErrs(rs.fromBkt.IsObjNotFoundErr).Get(ctx, path.Join(meta.ULID.String(), metadata.DeletionMarkFilename))
+		if err != nil && !rs.fromBkt.IsObjNotFoundErr(err) {
+			level.Debug(rs.logger).Log("msg", "failed to read deletion mark from origin bucket", "error", err)
```

Comment:
Is this an error, error? Or is this more like 'it could be there, which is an acceptable use-case'? - I'm questioning if this then needs to be Debug log level, info or perhaps even being a fatal error then?

Reply:
Right, the level here is disputable. The semantics is "I was not able to tell if the block is marked as deleted" and the question is what should be the behavior in such case.

This way could happen that we replicate a block marked for deletion even if the flag was enabled.  
I don't have a strong opinion here, maybe it would be safer to bail out completely with an error instead of possibly doing something that configuration tends to forbid.

Reply:
@GiedriusS WDYT?

Reply:
Mhm, yeah, maybe let's return an error instead of logging it? I remember there being a discussion about the same exact point but can't find anything now. Something along the lines of "crash early" :thinking: maybe we should add this to our code style document

Reply:
Ok, agreed changed in https://github.com/thanos-io/thanos/commit/b96326a0c66943a712bbd8189b59dbe8ad264e04 to fail with an error